Assets/Plugins/iOS/Shopify/PKError+JSON.swift

```diff
@@ -0,0 +1,141 @@
+    class func paymentError(with json: JSON) -> Error? {
+        
+        guard let errorType = ErrorType(json: json) else {
+            return nil
+        }
+        
+        switch errorType {
+        case .paymentBillingAddress:
+            return paymentBillingAddressInvalidError(from: json)
+        case .paymentShippingAddress:
+            return paymentShippingAddressInvalidError(from: json)
+        case .paymentContactInvalid:
+            return paymentContactInvalidError(from: json)
+        case .shippingAddressUnservicable:
+            return paymentShippingAddressUnserviceableError(from: json)
+        }
+    }
+}
+
+//  ----------------------------------
+//  MARK: - Fileprivate Error Creators -
+//
+@available(iOS 11.0, *)
+extension PKPaymentRequest {
+    
+    fileprivate class func paymentBillingAddressInvalidError(from json: JSON) -> Error? {
+
+        guard
+            let description        = json[PaymentErrorKey.description.rawValue] as? String,
+            let addressFieldString = json[PaymentErrorKey.field.rawValue] as? String,
+            let addressKey         = CNPostalAddress.postalAddressKey(for: addressFieldString)
+        else {
+            return nil
+        }
+
+        return paymentBillingAddressInvalidError(withKey: addressKey, localizedDescription: description)
+    }
+    
+    fileprivate class func paymentShippingAddressInvalidError(from json: JSON) -> Error? {
+
+        guard
+            let description        = json[PaymentErrorKey.description.rawValue] as? String,
+            let addressFieldString = json[PaymentErrorKey.field.rawValue] as? String,
+            let addressKey         = CNPostalAddress.postalAddressKey(for: addressFieldString)
+        else {
+            return nil
+        }
+
+        return paymentShippingAddressInvalidError(withKey: addressKey, localizedDescription: description)
+    }
+    
+    fileprivate class func paymentContactInvalidError(from json: JSON) -> Error? {
+
+        guard
+            let description        = json[PaymentErrorKey.description.rawValue] as? String,
+            let contactFieldString = json[PaymentErrorKey.field.rawValue] as? String,
+            let contactField       = PKContactField(string: contactFieldString)
+        else {
+            return nil
+        }
+
+        return paymentContactInvalidError(withContactField: contactField, localizedDescription: description)
+    }
+    
+    fileprivate class func paymentShippingAddressUnserviceableError(from json: JSON) -> Error? {
+
+        if let description = json[PaymentErrorKey.description.rawValue] as? String {
+            return paymentShippingAddressUnserviceableError(withLocalizedDescription: description)
+        } else {
+            return nil
+        }
+    }
+}
+
+// ----------------------------------
+//  MARK: - Parsing Errors -
+//
+@available(iOS 11.0, *)
+extension PKPaymentRequest {
+
+    fileprivate enum PaymentErrorKey: String {
```

Comment:
This doesn't look like the values are ever used for their `enum` types but rather for their underlying value. I personally prefer to represent keys as static constants:
```swift
private enum Keys {
    static let type        = "type"
    static let description = "description"
    static let field       = "field"
}
```
You can also "un-nest" them from `PKPaymentRequest` and convert to `private` to simplify naming since they are just simple JSON keys.